src/layer/highway_shield.js

```diff
@@ -44,38 +42,65 @@ let shieldLayout = {
   "symbol-placement": "line",
   "text-max-angle": 180,
   "text-pitch-alignment": "viewport",
+  "symbol-sort-key": [
+    "match",
+    ["get", "class"],
+    "motorway",
+    0,
+    "trunk",
+    1,
+    "primary",
+    2,
+    "secondary",
+    3,
+    "tertiary",
+    4,
+    5,
+  ],
 };
 
-let baseShield = {
+export const shield = {
   type: "symbol",
-  layout: shieldLayout,
   source: "openmaptiles",
-  metadata: {},
   "source-layer": "transportation_name",
-};
-
-function shieldLayer(hwyClass, minzoom) {
-  var layer = Util.cp(baseShield);
```

Comment:
Oops, looks like I wasn't using minzoom at all.  Which means it's not necessary :+1:

Reply:
Oop. The `text-opacity` value implements what I think was intended here.

Reply:
Well, nobody was missing it.